### ISSUE TITLE
Test fails even after successful compilation.

### ISSUE THREAD

**Issue by abhayraw1** [coin-or/Ipopt#385]
I followed the steps to install ipopt from [https://coin-or.github.io/Ipopt/INSTALL.html](https://coin-or.github.io/Ipopt/INSTALL.html)

I ran `./configure --with-hsl-lib` and its output follows:
```
checking build system type... x86_64-unknown-linux-gnu
checking whether we want to compile in debug mode... no
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
configure: C compiler options are: -O3 -pipe -DNDEBUG -Wimplicit -Wparentheses -Wsequence-point -Wreturn-type -Wcast-qual -Wall -Wno-unknown-pragmas -Wno-long-long
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether C++ compiler g++ works... yes
configure: C++ compiler options are: -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long
configure: Trying to determine Fortran compiler name
checking for gfortran... gfortran
checking whether we are using the GNU Fortran 77 compiler... yes
checking whether gfortran accepts -g... yes
configure: Fortran compiler options are: -O3 -pipe
checking for egrep... grep -E
checking whether ln -s works... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking dependency style of g++... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
checking host system type... x86_64-unknown-linux-gnu
checking for a sed that does not truncate output... /bin/sed
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for /usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking how to recognise dependent libraries... pass_all
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking how to run the C++ preprocessor... g++ -E
checking the maximum length of command line arguments... 32768
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC
checking if gcc PIC flag -fPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating libtool
appending configuration tag "CXX" to libtool
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC
checking if g++ PIC flag -fPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
appending configuration tag "F77" to libtool
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for gfortran option to produce PIC... -fPIC
checking if gfortran PIC flag -fPIC works... yes
checking if gfortran static flag -static works... yes
checking if gfortran supports -c -o file.o... yes
checking whether the gfortran linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
configure: Build is "x86_64-unknown-linux-gnu".
checking if library version is set... no
checking whether source of project ASL is available and should be compiled... no, source file ./ThirdParty/ASL/solvers/asl.h not available
checking whether source of project Metis is available and should be compiled... no, source file ./ThirdParty/Metis/metis-4.0/Lib/balance.c not available
checking whether source of project Blas is available and should be compiled... no, source file ./ThirdParty/Blas/daxpy.f not available
checking whether source of project Lapack is available and should be compiled... no, source file ./ThirdParty/Lapack/LAPACK/SRC/dlarf.f not available
checking whether source of project HSL is available and should be compiled... no, --with-hsl-lib has been specified
checking whether source of project Mumps is available and should be compiled... no, source file ./ThirdParty/Mumps/MUMPS/src/dmumps_part1.F not available
checking whether source of project Ipopt is available and should be compiled... yes, source in Ipopt
configure: configuring doxygen documentation options
checking for doxygen... no
checking for latex... no
checking for dot... YES
checking which command should be used to link input files... ln -s
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doxydoc/doxygen.conf
config.status: executing depfiles commands
configure: configuring in Ipopt
configure: running /bin/bash './configure' --prefix=/home/abhay/Ipopt  '--with-hsl-lib' 'LDFLAGS=-L/home/abhay/HSL/lib' --cache-file=/dev/null --srcdir=.
checking build system type... x86_64-unknown-linux-gnu
checking for svnversion... no
checking whether we want to compile in debug mode... no
checking for egrep... grep -E
checking whether ln -s works... yes
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
configure: C compiler options are: -O3 -pipe -DNDEBUG -Wimplicit -Wparentheses -Wsequence-point -Wreturn-type -Wcast-qual -Wall -Wno-unknown-pragmas -Wno-long-long   -DIPOPT_BUILD
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether C++ compiler g++ works... yes
configure: C++ compiler options are: -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long   -DIPOPT_BUILD
configure: Assuming that CXXLIBS is "-lstdc++ -lm".
configure: Trying to determine Fortran compiler name
checking for gfortran... gfortran
checking whether we are using the GNU Fortran 77 compiler... yes
checking whether gfortran accepts -g... yes
configure: Fortran compiler options are: -O3 -pipe
checking how to get verbose linking output from gfortran... -v
checking for Fortran libraries of gfortran...  -L/home/abhay/HSL/lib -L/usr/lib/gcc/x86_64-linux-gnu/7 -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/7/../../.. -lgfortran -lm -lgcc_s -lquadmath
configure: Corrected Fortran libraries:  -L/home/abhay/HSL/lib -L/usr/lib/gcc/x86_64-linux-gnu/7 -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/7/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/7/../../.. -lgfortran -lm -lquadmath
checking for dummy main to link with Fortran libraries... none
checking for Fortran name-mangling scheme... lower case, underscore, no extra underscore
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking dependency style of g++... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
configure: Using libtool script in directory ..
checking if library version is set... 11:11:10
configure: configuring doxygen documentation options
checking for doxygen... no
checking for latex... no
checking for dot... YES
checking for pkg-config... pkg-config
checking pkg-config is at least version 0.16.0... yes
checking whether -lblas has BLAS... yes: -lblas
checking whether LAPACK is already available with BLAS library... no
checking whether -llapack has LAPACK... yes: -llapack
checking for COIN-OR package ASL... not given: No package 'coinasl' found
checking for COIN-OR package HSL... yes
checking for COIN-OR package Mumps... not given: No package 'coinmumps' found
checking for int *... yes
checking size of int *... 8
configure: Using CXXAR="ar cru".
checking cmath usability... yes
checking cmath presence... yes
checking for cmath... yes
checking cctype usability... yes
checking cctype presence... yes
checking for cctype... yes
checking cassert usability... yes
checking cassert presence... yes
checking for cassert... yes
checking cstdio usability... yes
checking cstdio presence... yes
checking for cstdio... yes
checking cstdlib usability... yes
checking cstdlib presence... yes
checking for cstdlib... yes
checking cstdarg usability... yes
checking cstdarg presence... yes
checking for cstdarg... yes
checking cstring usability... yes
checking cstring presence... yes
checking for cstring... yes
checking cfloat usability... yes
checking cfloat presence... yes
checking for cfloat... yes
checking ctime usability... yes
checking ctime presence... yes
checking for ctime... yes
checking cstddef usability... yes
checking cstddef presence... yes
checking for cstddef... yes
checking for cmath... (cached) yes
checking for cfloat... (cached) yes
checking cieeefp usability... no
checking cieeefp presence... no
checking for cieeefp... no
checking ieeefp.h usability... no
checking ieeefp.h presence... no
checking for ieeefp.h... no
checking whether isfinite is declared... no
checking whether finite is declared... yes
checking whether va_copy is declared... yes
checking whether drand48 is declared... yes
checking whether rand is declared... yes
checking whether srand is declared... yes
checking whether MA28 is available with HSL... yes
checking for mkl_get_version_string... no
checking for MPI_Initialized... no
checking whether this is a VPATH configuration... no
checking for matlab installation... no
checking whether the linear solver loader should be compiled... yes
checking windows.h usability... no
checking windows.h presence... no
checking for windows.h... no
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking for dlopen in -ldl... yes
checking for snprintf... yes
checking for vsnprintf... yes
checking which command should be used to link input files... ln -s
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Common/Makefile
config.status: creating src/LinAlg/Makefile
config.status: creating src/LinAlg/TMatrices/Makefile
config.status: creating src/Interfaces/Makefile
config.status: creating src/Algorithm/Makefile
config.status: creating src/Algorithm/LinearSolvers/Makefile
config.status: creating src/Algorithm/Inexact/Makefile
config.status: creating src/contrib/CGPenalty/Makefile
config.status: creating src/contrib/LinearSolverLoader/Makefile
config.status: creating src/Apps/Makefile
config.status: creating src/Apps/AmplSolver/Makefile
config.status: creating src/Apps/CUTErInterface/Makefile
config.status: creating examples/hs071_f/Makefile
config.status: creating examples/hs071_f/hs071_f.f
config.status: creating examples/Cpp_example/Makefile
config.status: creating examples/hs071_cpp/Makefile
config.status: creating examples/hs071_c/Makefile
config.status: creating examples/ScalableProblems/Makefile
config.status: creating tutorial/CodingExercise/C/1-skeleton/Makefile
config.status: creating tutorial/CodingExercise/C/2-mistake/Makefile
config.status: creating tutorial/CodingExercise/C/3-solution/Makefile
config.status: creating tutorial/CodingExercise/Cpp/1-skeleton/Makefile
config.status: creating tutorial/CodingExercise/Cpp/2-mistake/Makefile
config.status: creating tutorial/CodingExercise/Cpp/3-solution/Makefile
config.status: creating tutorial/CodingExercise/Matlab/1-skeleton/startup.m
config.status: creating tutorial/CodingExercise/Matlab/2-mistake/startup.m
config.status: creating tutorial/CodingExercise/Matlab/3-solution/startup.m
config.status: creating tutorial/CodingExercise/Fortran/1-skeleton/TutorialFortran.f
config.status: creating tutorial/CodingExercise/Fortran/2-mistake/TutorialFortran.f
config.status: creating tutorial/CodingExercise/Fortran/3-solution/TutorialFortran.f
config.status: creating tutorial/CodingExercise/Fortran/1-skeleton/Makefile
config.status: creating tutorial/CodingExercise/Fortran/2-mistake/Makefile
config.status: creating tutorial/CodingExercise/Fortran/3-solution/Makefile
config.status: creating test/Makefile
config.status: creating test/run_unitTests
config.status: creating contrib/MatlabInterface/src/Makefile
config.status: creating contrib/MatlabInterface/examples/startup.m
config.status: creating contrib/JavaInterface/Makefile
config.status: creating contrib/sIPOPT/Makefile
config.status: creating contrib/sIPOPT/AmplSolver/Makefile
config.status: creating contrib/sIPOPT/examples/parametric_cpp/Makefile
config.status: creating contrib/sIPOPT/examples/parametric_dsdp_cpp/Makefile
config.status: creating contrib/sIPOPT/examples/redhess_cpp/Makefile
config.status: creating contrib/sIPOPT/src/Makefile
config.status: creating ipopt.pc
config.status: creating ipopt-uninstalled.pc
config.status: creating doxydoc/doxygen.conf
config.status: creating contrib/RInterface/src/Makevars
config.status: creating src/Common/config.h
config.status: src/Common/config.h is unchanged
config.status: creating src/Common/config_ipopt.h
config.status: src/Common/config_ipopt.h is unchanged
config.status: creating examples/ScalableProblems/config.h
config.status: examples/ScalableProblems/config.h is unchanged
config.status: linking ./examples/hs071_cpp/hs071_main.cpp to test/hs071_main.cpp
config.status: linking ./examples/hs071_cpp/hs071_nlp.cpp to test/hs071_nlp.cpp
config.status: linking ./examples/hs071_cpp/hs071_nlp.hpp to test/hs071_nlp.hpp
config.status: linking ./examples/hs071_c/hs071_c.c to test/hs071_c.c
config.status: executing depfiles commands
configure: In case of trouble, first consult the troubleshooting page at https://projects.coin-or.org/BuildTools/wiki/user-troubleshooting
configure: Configuration of Ipopt successful
configure: In case of trouble, first consult the troubleshooting page at https://projects.coin-or.org/BuildTools/wiki/user-troubleshooting
configure: Main configuration of Ipopt successful
```
This was followed by running `make` and `make test`
`make` didn't generate any errors as such but `make test` gives the following error:

```
Making all in Ipopt
make[1]: Entering directory '/home/abhay/Ipopt/Ipopt'
Making all in src/Common
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Common'
make  all-am
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Common'
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Common'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Common'
Making all in src/LinAlg
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
Making all in TMatrices
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg/TMatrices'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg/TMatrices'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
Making all in src/Algorithm
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
Making all in LinearSolvers
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/LinearSolvers'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/LinearSolvers'
Making all in Inexact
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/Inexact'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/Inexact'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
Making all in src/contrib/CGPenalty
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/contrib/CGPenalty'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/contrib/CGPenalty'
Making all in src/contrib/LinearSolverLoader
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/contrib/LinearSolverLoader'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/contrib/LinearSolverLoader'
Making all in src/Interfaces
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Interfaces'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Interfaces'
Making all in src/Apps
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps'
Making all in CUTErInterface
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps/CUTErInterface'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps/CUTErInterface'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt'
make[2]: Nothing to be done for 'all-am'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt'
make[1]: Leaving directory '/home/abhay/Ipopt/Ipopt'
make[1]: Entering directory '/home/abhay/Ipopt'
make[1]: Nothing to be done for 'all-am'.
make[1]: Leaving directory '/home/abhay/Ipopt'
cd Ipopt; make test
make[1]: Entering directory '/home/abhay/Ipopt/Ipopt'
Making all in src/Common
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Common'
make  all-am
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Common'
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Common'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Common'
Making all in src/LinAlg
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
Making all in TMatrices
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg/TMatrices'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg/TMatrices'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/LinAlg'
Making all in src/Algorithm
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
Making all in LinearSolvers
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/LinearSolvers'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/LinearSolvers'
Making all in Inexact
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/Inexact'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm/Inexact'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Algorithm'
Making all in src/contrib/CGPenalty
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/contrib/CGPenalty'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/contrib/CGPenalty'
Making all in src/contrib/LinearSolverLoader
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/contrib/LinearSolverLoader'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/contrib/LinearSolverLoader'
Making all in src/Interfaces
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Interfaces'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Interfaces'
Making all in src/Apps
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps'
Making all in CUTErInterface
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps/CUTErInterface'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps/CUTErInterface'
make[3]: Entering directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/src/Apps'
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt'
make[2]: Nothing to be done for 'all-am'.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt'
cd test; make test
make[2]: Entering directory '/home/abhay/Ipopt/Ipopt/test'
make[2]: *** No rule to make target 'yes', needed by 'hs071_cpp'.  Stop.
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/test'
Makefile:1063: recipe for target 'unitTest' failed
make[1]: *** [unitTest] Error 2
make[1]: Leaving directory '/home/abhay/Ipopt/Ipopt'
Makefile:694: recipe for target 'test' failed
make: *** [test] Error 2
```
I am trying this on Windows Subsystem for Linux with Ubuntu 18.04

**Comment by svigerske** [comment]
The `--with-hsl-lib` argument doesn't make any sense like this, but maybe it didn't do any harm.

I don't really see why it would try to make a target `yes` in test. Can you attach `/home/abhay/Ipopt/Ipopt/test/Makefile` and the output of `make test V=1`?

**Comment by abhayraw1** [comment]
I tried searching for any target named `yes` but its not there! Also If I don't use the `--with-hsl-lib` argument, I get a different error running `make test` which you can view [here](https://gist.github.com/abhayraw1/3502d788d8b12d9e20ef70b4969160b4). 
**TL;DR**
```
../src/Interfaces/.libs/libipopt.so: undefined reference to `mc68_default_control'
../src/Interfaces/.libs/libipopt.so: undefined reference to `mc68_order'
collect2: error: ld returned 1 exit status
Makefile:434: recipe for target 'hs071_cpp' failed
make[2]: *** [hs071_cpp] Error 1
make[2]: Leaving directory '/home/abhay/Ipopt/Ipopt/test'
Makefile:1063: recipe for target 'unitTest' failed
make[1]: *** [unitTest] Error 2
make[1]: Leaving directory '/home/abhay/Ipopt/Ipopt'
Makefile:694: recipe for target 'test' failed
make: *** [test] Error 2
```

Also the output of `./configure` has this line `checking whether source of project HSL is available and should be compiled... no, source file ./ThirdParty/HSL/coinhsl/common/deps.f not available` vs `checking whether source of project HSL is available and should be compiled... no, --with-hsl-lib has been specified` when specified with that argument.
I have a hunch that this could be the reason for the `yes` mystery!

You can find the Makefile file [here](https://gist.github.com/abhayraw1/d3e3f8947d8d1b215a4f78517f687235)

Also, I want to use ma57. I did received the academic license for the HSL solver and compiled them separately.
In the output of the `./configure` command there is this line `checking whether MA28 is available with HSL... yes`. How can I be sure that Ipopt will include the ma57 solver too? If that makes sense!

PS: The HSL library is installed in the `/usr/local/lib` directory. The output to `make test V=1` is included in the snap below

![image](https://user-images.githubusercontent.com/14311538/86955003-829e4900-c174-11ea-8cda-f228ef58693e.png)

**Comment by svigerske** [comment]
`--with-hsl-lib` expects the flags to link against the HSL library. Specifying none is something configure doesn't expect and doesn't check for, but it leads to thinking that the flags should be "yes". "yes" then [becomes a dependency of the test target](https://gist.github.com/abhayraw1/d3e3f8947d8d1b215a4f78517f687235#file-ipopt_test_makefile-L212) and since there is no file called yes, make complains.

If you don't specify `--with-hsl-lib`, you `libcoinhsl.so` [seems to be picked up correctly](https://gist.github.com/abhayraw1/3502d788d8b12d9e20ef70b4969160b4#file-make_test_op-L124), but it seems to be missing HSL_MC68. I cannot say why that is the case.

Since you seem to try to compile an old version of Ipopt (before 3.13), the installation instructions that you look at do not match exactly. With the old Ipopt, the easiest is to place the HSL source as obtained from http://www.hsl.rl.ac.uk/ipopt/ in the ThirdParty/HSL directory, so that `./ThirdParty/HSL/coinhsl/common/deps.f` can be found and then run the configure in the base directory. That should then compile both HSL and Ipopt.

MA27 would be the default, if present, but if MA57 is compiled as well, then the linear_solver option in Ipopt can be used to switch to MA57.

**Comment by abhayraw1** [comment]
Okay. I installed the latest release of Ipopt and all tests passed!
Is there any way I could change Ipopt to use ma57 instead of ma27 which is set by default?

**Comment by svigerske** [comment]
Yes, set the linear_solver option.
